rewrite-core/src/main/java/org/openrewrite/marker/GitProvenance.java

```diff
@@ -67,14 +75,40 @@ public class GitProvenance implements Marker {
     @Incubating(since = "8.9.0")
     List<Committer> committers;
 
+    @Incubating(since = "8.33.0")
+    @Nullable
+    transient CloneUrl cloneUrl;
+
+    public GitProvenance(UUID id,
+                         @Nullable String origin,
+                         @Nullable String branch,
+                         @Nullable String change,
+                         @Nullable AutoCRLF autocrlf,
+                         @Nullable EOL eol,
+                         @Nullable List<Committer> committers) {
+        this.id = id;
+        this.origin = origin;
+        this.branch = branch;
+        this.change = change;
+        this.autocrlf = autocrlf;
+        this.eol = eol;
+        this.committers = committers;
+        cloneUrl = parseCloneUrl(origin);
```

Comment:
We should probably only calculate this when it is first used, otherwise this slows down deserialization.